### ISSUE TITLE
`onMoveStart` doesn't work as expected

### ISSUE THREAD

**Issue by TitoGrine** [wbkd/react-flow#1452]
_First and foremost, great library, it has been a pleasure to use._

I believe the `onMoveStart` handler is not really working as intended, or at the very least its use seems very limited.

### Context

I'm trying to have the cursor style change when the user is panning (going from "grab" to "grabbing"). My though was to use the `onMove` handlers to trigger this change. So the flow would go something like this:
 1. `onMoveStart` - cursor style would be set to "grabbing"
 2. If the zoom changed and not the position, then set the style back to "grab"
 3. `onMoveEnd` - cursor style would be set to "grab"

The problem is that `onMoveStart` doesn't actually trigger at the start of a "movement" (be that zooming or panning).
I tried using the `onMove` instead but this is called too frequently so there is no chance for me to set the styling back to "grab" in time for it to matter.
The `onMoveEnd` handler works perfectly, but `onMoveStart` is only triggered once, with the first movement after render. 

Here is a [codesandbox](https://codesandbox.io/s/react-flow-onmovestart-own6m?file=/src/App.js) that illustrates this problem with `console.log`.

### The problem

When playing around with the code I found out that the "issue" is with the following code on `src/container/ZoomPane/index.tsx`:

```jsx
useEffect(() => {
  if (d3Zoom) {
    if (onMoveStart) {
      d3Zoom.on('start', (event: any) => {
        if (viewChanged(prevTransform.current, event.transform)) {
          const flowTransform = eventToFlowTransform(event.transform);
          prevTransform.current = flowTransform;

          onMoveStart(flowTransform);
        }
      });
    } else {
      d3Zoom.on('start', null);
    }
  }
}, [d3Zoom, onMoveStart]);
 ```
In order for the handler to trigger we must detect that something changed using `viewChanged`. At first glance this seems to make sense, we only want to trigger an `onMoveStart` event when we detect that movement happened. This prevents, for example, triggering this handler when the user simply clicks the viewport but doesn't actually pan or zoom.  
The logic fails because (I believe) d3-zoom triggers the `start` event before the movement actually begins, so, no matter what the user does, `prevTransform.current` and `event.transform` will always be the same when this is called and `event.transform` will only be changed afterwards. This explains why `onMoveEnd` works fine, because by the time it's called, the transformations are updated so it actually makes sense to call `viewChanged`.  
The reason why `onMoveStart` is triggered on the first movement after render is because `prevTransform.current` is first set to be `{ x: 0, y: 0, zoom: 0 }`, so there will be a difference in the beginning between this and `event.transform`.

### Conclusion

It may be that the intention behind `onMoveStart` is really to just trigger on the very first movement, but in my opinion it would make more sense to always have the following flow: `onMoveStart` → `onMove` → `onMoveEnd` → `onMoveStart` → ...
I think this is useful because I can imagine many scenarios where you want to trigger something when panning/zooming begins and reset it when it ends.  

Off the top of my head, I can think of two "solutions":
 1. Simply not check for `viewChanged` when dealing with `onMoveStart` and accept that it may be triggered when the user clicks but no actual movement happened;
 2. Instead of binding `onMoveStart` to the `start` event of d3-zoom, we bind it to `zoom` just like with `onMove` and have our own move state. Something like this (I didn't actually implement it):
```jsx
// Hook to keep state of movement
const [movStopped, setMovStopped] = useMovStopped(true);

...

useEffect(() => {
  if (d3Zoom) {
    if (onMoveStart) {
      d3Zoom.on('zoom.start', (event: any) => {
        if (movStopped) {
          const flowTransform = eventToFlowTransform(event.transform);
          onMoveStart(flowTransform);
          setMovStopped(false);
        }
      });
    } else {
      d3Zoom.on('zoom.start', null);
    }
  }
}, [d3Zoom, onMoveStart]);

useEffect(() => {
  if (d3Zoom) {
    if (onMoveEnd) {
      d3Zoom.on('end', (event: any) => {
        if (viewChanged(prevTransform.current, event.transform)) {
          const flowTransform = eventToFlowTransform(event.transform);
          prevTransform.current = flowTransform;

          onMoveEnd(flowTransform);
          setMovStopped(true);
        }
      });
    } else {
      d3Zoom.on('end', null);
    }
  }
}, [d3Zoom, onMoveEnd]);
```
I don't know d3-zoom well, so I can't think of anything else.  

Either way, it would be nice to address this if in fact `onMoveStart` is not working as intended. If the second solution seems viable I can try implementing it and open a PR.

Cheers :beers:

**Comment by nicolasletoublon** [comment]
I think all the same as the comment above :) 
All libraries like react-dnd or any moving library have the logic for: start-move-end-start-move-end :) 

Thanks for fixing that!

By the way, amazing work on this library!

**Comment by nicolasletoublon** [comment]
@moklick Do you have any insight on this please ?

**Comment by moklick** [comment]
Thanks @TitoGrine :) Fix released in v9.6.7
